app/lib/features/chat_ng/providers/chat_room_messages_provider.dart

```diff
@@ -49,3 +53,30 @@ final renderableChatMessagesProvider =
     return _supportedTypes.contains(msg.eventItem()?.eventType());
   }).toList();
 });
+
+// Provider to check if we should show avatar by comparing with the next message
+final shouldShowAvatarProvider = Provider.family<bool, RoomMsgId>(
```

Comment:
Nice idea of just moving the logic into a provider. Do you think we could have some tests for it? That'd be awesome, especially since it is pretty simple now but then could increase in complexity and we could always have test showing it works as wanted!

Reply:
Yes, they will be included in this PR :)